sdk/compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Upgrade.hs

```diff
@@ -111,44 +130,83 @@ checkBothAndSingle world checkBoth checkSingle version upgradeInfo mbUpgradedPac
     extractDiagnostics bothPkgDiagnostics ++ extractDiagnostics singlePkgDiagnostics
 
 checkUpgrade
-  :: LF.Package
+  :: LF.Package -> [LF.DalfPackage]
   -> Version -> UpgradeInfo
-  -> Maybe (LF.PackageId, LF.Package)
+  -> Maybe ((LF.PackageId, LF.Package), [(LF.PackageId, LF.Package)])
   -> [Diagnostic]
-checkUpgrade pkg =
+checkUpgrade pkg deps =
     let world = initWorldSelf [] pkg
-        checkBoth upgradedPkgId upgradedPkg =
-          checkUpgradeM upgradedPkgId (Upgrading upgradedPkg pkg)
+        checkBoth upgradedPkgId upgradedPkg upgradedDeps = do
+          deps <- checkUpgradeDependenciesM deps upgradedDeps
+          local (\env -> env { _upgradingDeps = deps }) $
```

Comment:
I noticed an opportunity to refactor this so that we don't have to set the dependencies to `[]` further up - now `checkBothAndSingle` initializes the UpgradingEnv's `deps` field, so that `checkBoth` doesn't need to - changes are in 9d3a67bb79 "Refactor checkBothAndSingle to initialize upgrade dependencies env"